Allow building with dynamic libssl

iczero opened the pull request:
The cmake build will always statically link `libssl` and `libcrypto`. In some cases, it may be preferable to dynamically link them. This adds `-DSSL_DYNAMIC=1` which will build `libxquic` dynamically linked to `libssl` and `libcrypto` and also disable `libxquic-static`.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/alibaba/xquic?pullRequest=389) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/alibaba/xquic?pullRequest=389) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/alibaba/xquic?pullRequest=389) it.</sub>

Kulsk commented on CMakeLists.txt:
```diff
@@ -58,15 +58,17 @@ endif()
+    message(NOTICE "building with dynamic libssl")
+    set(SSL_LIB_PATH ${SSL_LIBRARIES})
+else()
+    message(NOTICE, "building with static libssl")
```
remove char ',' after NOTICE

iczero commented:
Rebased and fixed the format problem.

Please note the "not found" message is no longer required. If the library is not found, cmake will print the following:

```
-- Could NOT find SSL (missing: SSL_LIBRARY_STATIC CRYPTO_LIBRARY_STATIC) 
CMake Error at CMakeLists.txt:63 (message):
  ssl module not found
```

Kulsk reviewed (approved):
LGTM
